tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

```diff
@@ -882,4 +882,73 @@ public function testJoinWithNonUniqueAliasThrowsException()
 
         $qb->getSQL();
     }
+    
+    /**
+     * this test fails while it should not
+     */
+    public function testDoubleFromWithJoin()
+    {
+        $qb = new QueryBuilder($this->conn);
+
+        $qb->select('a.id, b.id, c.id')
+            ->from('table_a', 'a')
+            ->from('table_b', 'b')
+            ->join('a', 'table_c', 'c', 'a.fk_c = c.id')
+            ->where('a.fk_b = b.id');
+
+        $this->assertSame('SELECT a.id, b.id, c.id FROM table_a a, table_b b INNER JOIN table_c c ON a.fk_c = c.id WHERE a.fk_b = b.id', $qb->getSQL());
```

Comment:
From docs:
"A join always belongs to one part of the from clause. This is why you have to specify the alias of the FROM part the join belongs to as the first argument."
Why would the join cluase use table_b as base table if I specify 'a' alias?